### DISCUSSION TITLE
Unknown Recipients get mapped to ONE already existing Recipient

### DISCUSSION THREAD

**Discussion by numerfolt** [firefly-iii/firefly-iii#7739] [Firefly III] [answered]
Hey,
thanks for firefly-iii. The documentation is really good so that my setup was really easy.
I am selfhosting and use both firefly-iii and the data-importer. Now I dont want to map all accounts by hand all the time and was happy that the importer interface offered me to map my transactions.

But since then it didnt offer it anymore and all unknown recipients get mapped to "Marktkauf REDACTED" and I need to change them manually.

Now I am using auto-import with a config file that uses the "mapping" array like this (included "roles" in case it may help):
```
    "roles": [
        "date_book",
        "date_transaction",
        "note",
        "opposing-name",
        "description",
        "opposing-iban",
        "opposing-bic",
        "amount",
        "sepa_ct_op",
        "sepa_db",
        "_ignore",
        "_ignore"
    ],
    "do_mapping": [
    ],
    "mapping": [
        {
            "AESCULAP-APOTHEKE": 32,
            "ALDI SAGT DANKE": 35,
            "AMAZON EU S.A R.L., NIEDERLASSUNG DEUTSCHLAND": 15,
            "AMAZON PAYMENTS EUROPE S.C.A.": 13,
            "REDACTED": 11,
            "REDACTED E.V.": 18,
            "REDACTED": 20,
            "EDEKA REDACTED": 31,
            "REDACTED GMBH+CO.KG": 16,
            "FRAMEWORK. R647440317": 23,
            "Friseur REDACTED": 30,
            "H.M REDACTED 1DE0368": 36,
            "REDACTED": 10,
            "Jysk SE": 29,
            "KICKSTARTER.COM": 25,
            "MARKTKAUF REDACTED": 5,
            "REDACTED": 28,
            "NEW YORKER DEUTSCHLAND": 37,
            "PayPal (Europe) S.a r.l. et Cie, S.C.A.": 6,
            "PayPal Europe S.a.r.l. et Cie S.C.A": 8,
            "REDACTED GmbH": 17,
            "REDACTED": 26,
            "REDACTED E.V.": 12,
            "REDACTED": 19,
            "REDACTED": 34,
            "REDACTED a.G.": 14,
            "Telekom Deutschland GmbH                                              Landgrabenweg 151": 21,
            "REDACTED": 7,
            "Woolworth GmbH REDACTED": 33,
            "REDACTED": 27,
            "REDACTED GmbH": 9,
            "toom BM REDACTED": 24
        }
    ],
```

Because financial data is quite private, I redacted a lot of different mappings. The one Recipient that all the unknown Recipients are mapped to is the `"MARKTKAUF REDACTED": 5,` one.

I am customer of the DKB Giro model and tried to get some inspiration from this config file: https://github.com/firefly-iii/import-configurations/blob/main/de/dkb/csv-giro.json

Am I using the mapping config correctly or did I miss something?
And how would I handle new recipients to be automatically added to this array?

Thanks in advance!

**Comment by JC5** [comment]
Well the mapping interface being gone was a bug it should be back already. So there's that.

**Reply by numerfolt** [reply]
Ok, good to know. (I am on version 6.0.16 and 1.3.1)
How can I map the data with only automatic imports?
Or will I need to import manually every time a new recipient is in my transactions?

**Reply by JC5** [reply]
No, expanding the list of mappings should work just fine. However, could you verify the content of this array?

```json
    "do_mapping": [
    ],
```

Is it really empty, or did you remove it for clarity?

**Reply by numerfolt** [reply]
I left it empty, that's correct.

"No, expanding the list of mappings should work just fine."
Ok, great :)
I am not quite sure I understood how that works though...
Could you specify:
Do I need to expand the list of mappings manually by using the data importer?
Or
Does the list of mappings expand automatically when I add a new expense account in firefly-iii?

**Reply by JC5** [reply, marked as answer]
In the UI, the list will expand automatically, but if you import over the command line you must edit the list yourself.

> I am not quite sure I understood how that works though...

Add entries in this format:

```
"Name of expense account according to Nordigen", 123,
```

123 is the expense account ID in Firefly III. See the URL (`/accounts/show/x`).

One thing you must still do, and that is change the `do_mapping` array.

```
    "roles": [
        "date_book",
        "date_transaction",
        "note",
        "opposing-name",
        "description",
        "opposing-iban",
        "opposing-bic",
        "amount",
        "sepa_ct_op",
        "sepa_db",
        "_ignore",
        "_ignore"
    ],
    "do_mapping": [
false, false, false, true, false, false, false, false, false, false, false, false, 
    ],
```
It tels the data importer to use the mapping.

**Reply by numerfolt** [reply]
Thank you very much for your explanation!

I'll stick to the UI then 👍

Edit: The last comma in the "do_mapping" array is causing a syntax error, if anyone else reads this, make sure to remove the last comma.
